cmd/otelcol/main.go

```diff
@@ -36,10 +37,14 @@ import (
 )
 
 func main() {
+	runFromCmdLine(os.Args)
```

Comment:
This extra function is to make the code easily callable from tests without having to touch `os.Args` too much, notice that the actual code in fact already changes `os.Args`, but, I don't want to do that here when we can pass the arguments as a slice in the tests.

Reply:
sounds good